tests/test_rules.py

```diff
@@ -1,40 +1,56 @@
+from abc import ABCMeta, abstractmethod
 from unittest.mock import patch
 
 from django.test import TestCase
 
 from djangocms_fil_permissions.rules import has_site_access
 from djangocms_fil_permissions.test_utils.factories import (
     PollFactory,
+    RestaurantFactory,
     SiteFactory,
     UserFactory,
     UserSiteFactory,
 )
 
 
-class RulesTestCase(TestCase):
+class BaseRulesTests(metaclass=ABCMeta):
+    @abstractmethod
+    def factory(self, site):
+        pass
+
     def test_has_site_access(self):
         usersite = UserSiteFactory()
-        poll = PollFactory(site=usersite.site)
+        obj = self.factory(site=usersite.site)
 
-        self.assertTrue(has_site_access(usersite.user, poll))
+        self.assertTrue(has_site_access(usersite.user, obj))
 
     def test_has_site_access_no_access(self):
         user = UserFactory()
         site = SiteFactory()
-        poll = PollFactory(site=site)
+        obj = self.factory(site=site)
 
-        self.assertFalse(has_site_access(user, poll))
+        self.assertFalse(has_site_access(user, obj))
 
     def test_has_site_access_no_site_relation(self):
         user = UserFactory()
         site = SiteFactory()
-        poll = PollFactory(site=site)
+        obj = self.factory(site=site)
 
         with patch(
-            "djangocms_fil_permissions.rules.get_site_for_obj", return_value=None
-        ) as get_site_for_obj, patch(
-            "djangocms_fil_permissions.rules.user_has_access_to_site"
-        ) as user_has_access_to_site:
-            self.assertTrue(has_site_access(user, poll))
-        get_site_for_obj.assert_called_once_with(poll)
-        user_has_access_to_site.assert_not_called()
+            "djangocms_fil_permissions.rules.get_sites_for_obj", return_value=None
+        ) as get_sites_for_obj, patch(
+            "djangocms_fil_permissions.rules.user_can_access_any_of_sites"
+        ) as user_can_access_any_of_sites:
+            self.assertTrue(has_site_access(user, obj))
+        get_sites_for_obj.assert_called_once_with(obj)
+        user_can_access_any_of_sites.assert_not_called()
+
+
+class FKRulesTestCase(BaseRulesTests, TestCase):
+    def factory(self, site):
+        return PollFactory(site=site)
+
+
+class M2MRulesTestCase(BaseRulesTests, TestCase):
+    def factory(self, site):
+        return RestaurantFactory(sites=[site])
```

Comment:
Ditto on this test case I think. Probably should have the three additional test cases mentioned above.